Kickstarter-iOS/Views/Controllers/DiscoveryPageViewController.swift

```diff
@@ -87,26 +92,21 @@ internal final class DiscoveryPageViewController: UITableViewController {
     self.viewModel.inputs.viewDidDisappear(animated: animated)
   }
 
-  internal override func viewDidLayoutSubviews() {
-    super.viewDidLayoutSubviews()
-
-    self.loadingIndicatorView.center = self.tableView.center
-  }
-
   internal override func bindStyles() {
     super.bindStyles()
 
     _ = self
       |> baseTableControllerStyle(estimatedRowHeight: 200.0)
-
-    _ = self.loadingIndicatorView
-      |> baseActivityIndicatorStyle
   }
 
     internal override func bindViewModel() {
     super.bindViewModel()
 
-    self.loadingIndicatorView.rac.animating = self.viewModel.outputs.projectsAreLoading
+    self.viewModel.outputs.projectsAreLoading
+      .observeForUI()
+      .observeValues { [weak self] in
+        $0 ? self?.refreshControl?.beginRefreshing() : self?.refreshControl?.endRefreshing()
```

Comment:
I think we'll also have to pass back one more attribute (or use a different signal) for when this is manually triggered refresh.

manual = user pulled down to refresh
automatic = first time load of the page

Otherwise there's a weird bounce animation when the loading is done and the page control is being hidden. 

On the left - old behaviour | On the right new behaviour
![Untitled](https://user-images.githubusercontent.com/387596/54464942-105f5500-4736-11e9-947b-d57de4350ad4.gif)

There is a workaround for this:

```
self.viewModel.outputs.projectsAreLoading
  .observeForUI()
  .observeValues { [weak self] (isLoading, animated) in
    if isLoading {
      self?.performClosureAnimated(animated, closure: { self?.refreshControl?.endRefreshing() })
    } else {
      self?.performClosureAnimated(animated, closure: { self?.refreshControl?.beginRefreshing() })
    }
}
```

where we have a private function

```
private func performClosureAnimated(_ animated: Bool, closure: () -> Void) {
  if animated {
    closure()
  } else {
    UIView.performWithoutAnimation { closure() }
  }
}
```

Reply:
This one is a little trickier because is involves multiple signals. Will have to take some time to work on this one.